dot/utils_test.go

```diff
@@ -0,0 +1,607 @@
+	require.NoError(t, err)
+
+	filename = filepath.Join(t.TempDir(), "genesis.json")
+	err = os.WriteFile(filename, b, os.ModePerm)
+	require.NoError(t, err)
+
+	return filename
+}
+
+func TestCreateJSONRawFile(t *testing.T) {
+	type args struct {
+		bs *BuildSpec
+		fp string
+	}
+	tests := []struct {
+		name         string
+		args         args
+		expectedHash string
+	}{
+		{
+			name: "working example",
+			args: args{
+				bs: &BuildSpec{genesis: NewTestGenesis(t)},
+				fp: filepath.Join(t.TempDir(), "/test.json"),
+			},
+			expectedHash: "23356cdb5d3537d39b735726707216c9e329c7b8a2c8a41b25da0f5f936b3caa",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := CreateJSONRawFile(tt.args.bs, tt.args.fp)
+			file, err := os.Open(got.Name())
+			require.NoError(t, err)
+			defer file.Close()
+
+			buf := make([]byte, 30*1024)
+			sha256 := sha256.New()
+			for {
+				n, err := file.Read(buf)
+				if n > 0 {
+					_, err := sha256.Write(buf[:n])
+					require.NoError(t, err)
+				}
+				if err == io.EOF {
+					break
+				}
+				if err != nil {
+					log.Errorf("read %d bytes: %v", n, err)
+					break
+				}
+			}
+			sum := sha256.Sum(nil)
+			require.Equal(t, tt.expectedHash, fmt.Sprintf("%x", sum))
+		})
+	}
+}
+
+func TestExportConfig(t *testing.T) {
+	filepath := t.TempDir() + "/test.json"
+	type args struct {
+		cfg *Config
+		fp  string
+	}
+	tests := []struct {
+		name          string
+		args          args
+		want          *os.File
+		wantedContent string
+	}{
+		{
+			name: "working example",
+			args: args{
+				cfg: &Config{},
+				fp:  filepath,
+			},
+			want: &os.File{},
+			wantedContent: `[global]
+name = ""
+id = ""
+base_path = ""
+log_lvl = 0
+publish_metrics = false
+metrics_port = 0
+no_telemetry = false
+telemetry_urls = []
+retain_blocks = 0
+pruning = ""
+
+[log]
+core_lvl = 0
+digest_lvl = 0
+sync_lvl = 0
+network_lvl = 0
+rpc_lvl = 0
+state_lvl = 0
+runtime_lvl = 0
+block_producer_lvl = 0
+finality_gadget_lvl = 0
+
+[init]
+genesis = ""
+
+[account]
+key = ""
+unlock = ""
+
+[core]
+roles = 0
+babe_authority = false
+b_a_b_e_lead = false
+grandpa_authority = false
+wasm_interpreter = ""
+grandpa_interval = 0
+
+[network]
+port = 0
+bootnodes = []
+protocol_id = ""
+no_bootstrap = false
+no_m_dns = false
+min_peers = 0
+max_peers = 0
+persistent_peers = []
+discovery_interval = 0
+public_ip = ""
+public_dns = ""
+
+[rpc]
+enabled = false
+external = false
+unsafe = false
+unsafe_external = false
+port = 0
+host = ""
+modules = []
+w_s_port = 0
+w_s = false
+w_s_external = false
+w_s_unsafe = false
+w_s_unsafe_external = false
+
+[system]
+system_name = ""
+system_version = ""
+
+[state]
+rewind = 0
+
+[pprof]
+enabled = false
+
+[pprof.settings]
+listening_address = ""
+block_profile_rate = 0
+mutex_profile_rate = 0
+`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got := exportConfig(tt.args.cfg, tt.args.fp)
+			if tt.want != nil {
```

Comment:
This is always not nil, remove this field

Reply:
Removed.